### PR TITLE
Update go version to 1.18

### PR DESCRIPTION
**Pull request by Shikachuu**
## Description
Update go to version 1.18
## Motivation and Context

- [X] I have raised an issue to propose this change ([required](https://github.com/openfaas/faas/blob/master/CONTRIBUTING.md)) #7 
- [X] My issue has received approval from the maintainers or lead with the `design/approved` label

## How Has This Been Tested?
Run go build + make test

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [X] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [X] I've read the [CONTRIBUTION](https://github.com/openfaas/faas/blob/master/CONTRIBUTING.md) guide
- [X] I have signed-off my commits with `git commit -s`
- [ ] I have added tests to cover my changes.
- [X] All new and existing tests passed.

### CODE REVIEW

**Review by alexellis — APPROVED** [review]
Approved

File: Dockerfile [hunk 1]
```diff
@@ -1,6 +1,6 @@
 FROM teamserverless/license-check:0.3.9 as license-check
```
Comment (alexellis) [inline Dockerfile]: This also can be updated now to https://github.com/openfaas/license-check/pkgs/container/license-check and 0.4.0

File: go.mod [hunk 2]
```diff
@@ -1,10 +1,14 @@
 module github.com/openfaas/mqtt-connector
 
-go 1.15
+go 1.18
```
Comment (alexellis) [inline go.mod]: The go.mod version isn't important to update beyond 1.17 IMHO. The reason being is that setting it to 1.18 forces anyone referencing any of the code in this module to also use Go 1.18.

File: go.mod [hunk 3]
```diff
@@ -1,10 +1,14 @@
...
 
 require (
 	github.com/eclipse/paho.mqtt.golang v1.2.0
 	github.com/openfaas/connector-sdk v0.6.1
 	github.com/openfaas/faas-provider v0.17.1
+)
+
+require (
+	github.com/pkg/errors v0.9.1 // indirect
```
Comment (alexellis) [inline go.mod]: I'd like to remove all usage of errors.Wrap and change it over for this, if we have that code in our own files in this project.

```golang
fmt.Errorf("some additional context, error: %w", err)
```

Could you do that too?

File: go.mod [hunk 4]
```diff
@@ -1,10 +1,14 @@
 module github.com/openfaas/mqtt-connector
 
-go 1.15
+go 1.18
 
 require (
 	github.com/eclipse/paho.mqtt.golang v1.2.0
 	github.com/openfaas/connector-sdk v0.6.1
 	github.com/openfaas/faas-provider v0.17.1
```
Comment (alexellis) [inline go.mod]: faas-provider can be upgraded, as can the connector-sdk. Could you do those too?

File: main.go [hunk 5]
```diff
@@ -135,7 +136,7 @@ func main() {
 
 			log.Printf("Invoking (%s) on topic: %q, value: %q", gatewayURL, topic, data)
 
-			controller.Invoke(topic, &data)
+			controller.Invoke(topic, &data, http.Header{})
```
Comment (Shikachuu) [inline main.go:139]: @alexellis Tbh I'm not really sure about this, I do not fully understand how and why we need to pass headers there, this call here is fine or should I change something?

File: go.mod (hunk 4 above)
Comment (Shikachuu) [reply go.mod]: Every dependency here is now up-to-date.

File: main.go (hunk 5 above)
Comment (alexellis) [reply main.go:139]: In another PR we could consider what X-Key=Value kind of headers we could pass from the MQTT message like the originating server name for instance? Or anything else that might be useful as a recipient of a message from this connector

**Pull request merged**